src/common/scan-info/src/scan_operator.rs

```diff
@@ -0,0 +1,79 @@
+use std::{
+    any::Any,
+    fmt::{Debug, Display},
+    hash::{Hash, Hasher},
+    sync::Arc,
+};
+
+use common_error::DaftResult;
+use daft_schema::schema::SchemaRef;
+
+use crate::{PartitionField, Pushdowns};
+
+pub trait ScanTaskLike {
+    fn as_any(&self) -> &dyn Any;
+    fn as_any_arc(self: Arc<Self>) -> Arc<dyn Any + Send + Sync>;
+}
```

Comment:
I thought we'd have a bit more methods on here. I was thinking we'd define the entire interface of what a `ScanTask` is here, then impl it elsewhere. 

There are only a [handful](https://github.com/universalmind303/Daft/blob/2b71ffb7cc6cd42129e287cc14cc91a4c580c1b4/src/daft-scan/src/lib.rs#L384) of methods on the scan task, so i feel like it'd be pretty easy to put that all into the trait.

Reply:
I did this to make the actual API changes in this PR minimal, but I think it does make sense to define this trait a little better. Will work on that!